codec/src/main/java/io/netty/handler/codec/compression/PluggableByteBufOutputStream.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.internal.ObjectUtil;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * This is a special {@link OutputStream} implementation that
+ * allows {@link ByteBuf} to be plugged-in for writing data
+ * into it.
+ */
+class PluggableByteBufOutputStream extends OutputStream {
```

Comment:
Why not use `ByteBufOutputStream` ?

Reply:
We cannot change `ByteBuf` where compressed data is written during encoding. Storing all compressed in one single buffer is not a solution here. 

Though, I am all ears for improving this.

Reply:
There are ways where we don't have to use OutputStream at all but that requires modification in Brotli4j code a lot.